tests/src/python/test_qgsvectorlayerutils.py

```diff
@@ -258,14 +258,16 @@ def testCreateFeature(self):
         # layer with default value expression
         layer.setDefaultValueDefinition(2, QgsDefaultValue('3*4'))
         f = QgsVectorLayerUtils.createFeature(layer)
-        self.assertEqual(f.attributes(), [NULL, NULL, 12.0])
-        # we expect the default value expression to take precedence over the attribute map
+        self.assertEqual(f.attributes(), ['a', NULL, 6.0])
```

Comment:
I think this change is unintentional -- the values should not be affected by this bug fix

Reply:
Because of the new behavior, since we are using a value map as input, it should not be changed. Before, the default value had to overwrite the provided one, now the provided valye must remain the same, don't you agree?